### DISCUSSION TITLE
[prosemirror-model]: textBetween leafText can now be a function

### DISCUSSION THREAD

**Discussion by nathggns** [DefinitelyTyped/DefinitelyTyped#57769] [Issues with a @types package]
In 1.15.0 (2021-10-25) of prosemirror-model, `textBetween` (a common method on `Fragment` and `Node`, and maybe elsewhere) was updated to take a function for its `leafText` argument. In the event a function is passed, this function is passed the leaf node that needs to be converted to text, and is expected to return the leaf text for that node.

See:

- https://prosemirror.net/docs/changelog/ 
- https://prosemirror.net/docs/ref/#model.Node.textBetween
- https://prosemirror.net/docs/ref/#model.Fragment.textBetween

[a comment by an automated account is not shown]

**Comment by nathggns** [comment]
A workaround is this

```
interface ExtendedFragment<S extends Schema = any> extends Fragment<S> {
  textBetween(
    from: number,
    to: number,
    blockSeparator?: string | null,
    leafText?:
      | string
      | null
      | ((leafNode: ProsemirrorNode<S>) => string | null),
  ): string;
}
```

and then just cast to `ExtendedFragment` before using `textBetween`

**Comment by kachkaev** [comment]
This problem was fixed in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/58595. All `@types/prosemirror-*` package were later removed in favor of built-in Prosemirror typings: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/62888
